pkg/operator/baremetal_test.go

```diff
@@ -375,7 +378,163 @@ func TestSyncBaremetalControllers(t *testing.T) {
 
 			err := optr.syncBaremetalControllers(operatorConfig, tc.configCRName)
 			g.Expect(err).To(Equal(tc.expectedError))
+		})
+	}
+}
+
+func TestCheckMetal3DeploymentOwned(t *testing.T) {
+	kubeClient := fakekube.NewSimpleClientset(nil...)
+	operatorConfig := newOperatorWithBaremetalConfig()
+	client := kubeClient.AppsV1()
 
+	testCases := []struct {
+		testCase      string
+		deployment    *appsv1.Deployment
+		expected      bool
+		expectedError bool
+	}{
+		{
+			testCase: "Only maoOwnedAnnotation",
```

Comment:
Checking these annotations in tests and not checking them in real code is not useful.  In any case, I don't think we need the new CBO annotation in any case, at least not in this code base.

Reply:
maoOwnedAnnotation was added as part of https://github.com/openshift/machine-api-operator/pull/424. Only cboOwnedAnnotation is added as part of this change. This annotation will not be added to the metal3 deployment by any component in 4.6. It will be added by CBO in 4.7. During some part of the 4.7 dev cycle, both MAO and CBO would be capable of managing the metal3 deployment and during that time, we want CBO to take precedence.

Reply:
There's nothing in the actual product code that checks for this condition, and these tests don't call any MAO function that might result in both being present.  So, these tests are unnecessary.

Reply:
As @dhellmann explained it better than I could have:
 "The 4.6 and 4.7 configurations for metal3 will be different. Therefore if we ever have a time when 2 components are both trying to update the Deployment, they're going to thrash making changes back and forth. There are at least 2 situations where that could come up."

In this scenario where MAO and CBO are both trying to make changes to metal3 Deployment, MAO would have added the maoOwnedAnnotation to metal3 deployment with configurationA. And, CBO would also try to update the Deployment with configurationB. In addition, CBO would add the cboOwnedAnnotation to its own metal3 Deployment. With the changes added to syncBaremetalControllers() we will allow CBO to take control and MAO will back off.

Yes, no MAO function will result in both these annotation to be set. But, MAO will have to react correctly to situation where both annotations are set. Hence the test.